config/config.go

```diff
@@ -0,0 +1,60 @@
+/* validate that yaml conforms to the spec
+* Include logging about which buckets we end up nuking
+ */
+package config
+
+import (
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"path/filepath"
+  "regexp"
+)
+
+type RawConfig struct {
+  ResourceType resourceType `yaml:"s3"`
+}
+
+type resourceType struct {
+  MatchingRule []string `yaml:"include_names_regex"`
+}
+
+type ConfigObj struct {
+  S3FilterRule filterRule `yaml:"s3"`
+}
+
+type filterRule struct {
+  IncludeNamesRE []*regexp.Regexp `yaml:"include_names_regex"`
+}
```

Comment:
NIT: the `yaml` tags are unnecessary if we aren't directly marshalling/unmarshalling from this object.

Reply:
Does it not count as directly unmarshalling if we're unmarshalling into `rawConfig` [here](https://github.com/gruntwork-io/cloud-nuke/blob/2bd89a46ae9ffffd793fc406a42d2dfccd5e3a88/config/config.go#L57)?

Reply:
That counts if `filterRule` was somehow nested into the `RawConfig` struct, but looking at the code, `ConfigObj` looks disjoint from the `RawConfig` struct. So you need the tags on `RawConfig` and `resourceType` (which are being unmarshalled from the yaml), but I don't think you need it for `ConfigObj` and `filterRule` (which are being manually constructed in the routine). Does that make sense?

Reply:
Ah, yes, that makes sense. Removed the tags from the ConfigObj and descendents.